src/js/Content/Features/Community/Inventory/FInventoryMarketHelper.js

```diff
@@ -152,70 +152,88 @@ export default class FInventoryMarketHelper extends Feature {
         }
     }
 
-    _setBackgroundOption(thisItem, assetId, itemActions) {
+    async _setBackgroundOption(thisItem, itemActions) {
 
-        if (!document.querySelector(".inventory_links")) { return; }
         if (itemActions.querySelector(".es_set_background")) { return; }
 
+        // Make sure the selected item is a background
         const viewFullBtn = itemActions.querySelector("a");
-        if (!viewFullBtn) { return; }
+        if (!viewFullBtn || !/public\/images\/items/.test(viewFullBtn.href)) { return; }
 
-        if (!/public\/images\/items/.test(viewFullBtn.href)) { return; }
+        // Get owned backgrounds and the communityitemid for equipped background
+        if (!this.profileBgsOwned) {
+
+            // Retrieve access token from user's profile to use in Steam API calls
+            const html = await RequestData.getHttp(User.profileUrl);
+            const dummyPage = HTMLParser.htmlToDOM(html);
+            const config = dummyPage.getElementById("application_config").dataset.loyaltystore;
+            this.userToken = JSON.parse(config).webapi_token;
+
+            try {
+                const [equipped, owned] = await Promise.all([
+                    RequestData.getJson(`https://api.steampowered.com/IPlayerService/GetProfileItemsEquipped/v1?access_token=${this.userToken}&steamid=${User.steamId}`, {"credentials": "omit"}),
```

Comment:
Is it worth it to fetch this just to prevent setting it again?

Reply:
Currently we disable the button to indicate the background has been successfully changed, so I think it makes sense to also disable the button if the background had been set already.

Reply:
But this check is performed when a user clicks on the button, not before.
Also, if we unconditionally set the background, the button will be disabled anyway.

Reply:
It is also checked before inserting the button HTML: https://github.com/tfedor/AugmentedSteam/pull/1023/files#diff-34841fbfcec4ea7f01d735611484ec226141c8d96addbb173a3411cb44466a69R203-R204, so if the selected backgound is currently equipped, the button won't be clickable. IMO this is better than the old behaviour, it feels odd if I equip a background, reload the inventory and the button is clickable again.